explorer/src/Pos/Explorer/ExplorerMode.hs

```diff
@@ -0,0 +1,331 @@
+    dbGet = DB.dbGetPureDefault
+    dbIterSource = DB.dbIterSourcePureDefault
+
+instance HasConfigurations => DB.MonadDB ExplorerTestInitMode where
+    dbPut = DB.dbPutPureDefault
+    dbWriteBatch = DB.dbWriteBatchPureDefault
+    dbDelete = DB.dbDeletePureDefault
+
+instance
+    HasConfigurations =>
+    DB.MonadBlockDBGeneric BlockHeader Block Undo ExplorerTestInitMode
+  where
+    dbGetBlock  = DB.dbGetBlockPureDefault
+    dbGetUndo   = DB.dbGetUndoPureDefault
+    dbGetHeader = DB.dbGetHeaderPureDefault
+
+instance
+    HasConfigurations =>
+    DB.MonadBlockDBGenericWrite BlockHeader Block Undo ExplorerTestInitMode
+  where
+    dbPutBlund = DB.dbPutBlundPureDefault
+
+instance
+    HasConfigurations =>
+    DB.MonadBlockDBGeneric (Some IsHeader) SscBlock () ExplorerTestInitMode
+  where
+    dbGetBlock  = DB.dbGetBlockSscPureDefault
+    dbGetUndo   = DB.dbGetUndoSscPureDefault
+    dbGetHeader = DB.dbGetHeaderSscPureDefault
+
+----------------------------------------------------------------------------
+-- Boilerplate ExplorerTestContext instances
+----------------------------------------------------------------------------
+
+instance GS.HasGStateContext ExplorerTestContext where
+    gStateContext = etcGState_L
+
+instance HasSlottingVar ExplorerTestContext where
+    slottingTimestamp = etcSystemStart_L
+    slottingVar = GS.gStateContext . GS.gscSlottingVar
+
+instance HasLens DB.DBPureVar ExplorerTestContext DB.DBPureVar where
+    lensOf = GS.gStateContext . GS.gscDB . pureDBLens
+      where
+        getter = \case
+            DB.RealDB _   -> realDBInTestsError
+            DB.PureDB pdb -> pdb
+        setter _ pdb = DB.PureDB pdb
+        pureDBLens = lens getter setter
+        realDBInTestsError = error "You are using real db in tests"
+
+-- We need to define the full transformer stack type.
+type instance MempoolExt ExplorerExtraTestMode = ExplorerExtra
+
+instance HasLens TxpHolderTag ExplorerTestContext (GenericTxpLocalData ExplorerExtra) where
+    lensOf = etcTxpLocalData_L
+
+instance HasLens LoggerName ExplorerTestContext LoggerName where
+      lensOf = etcLoggerName_L
+
+instance HasLoggerName' ExplorerTestContext where
+    loggerName = lensOf @LoggerName
+
+instance HasJsonLogConfig ExplorerTestContext where
+    jsonLogConfig = jsonLogConfig
+
+----------------------------------------------------------------------------
+-- Boilerplate ExplorerTestMode instances
+----------------------------------------------------------------------------
+
+instance HasConfigurations => MonadGState ExplorerTestMode where
+    gsAdoptedBVData = DB.gsAdoptedBVDataDefault
+
+instance (HasConfigurations, MonadSlotsData ctx ExplorerTestMode)
+      => MonadSlots ctx ExplorerTestMode
+  where
+    getCurrentSlot = do
+        view etcSlotId_L >>= \case
+            Nothing -> Slot.getCurrentSlotSimple =<< view etcSSlottingVar_L
+            Just slot -> pure (Just slot)
```

Comment:
`maybe` or some alternative other then case look like a more readable alternative (to me at least, its subjective). Minor - and irrelevant